File: app/src/main/java/com/zsk/androtweet/Adapters/TweetAdapter.java
package com.zsk.androtweet.Adapters;

import android.content.Context;
import android.text.format.DateUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.CompoundButton.OnCheckedChangeListener;
import android.widget.ImageView;
import android.widget.TextView;

import com.zsk.androtweet.AndroTweetApp;
import com.zsk.androtweet.Main;
import com.zsk.androtweet.Models.Tweet;
import com.zsk.androtweet.R;

import java.util.List;

public class TweetAdapter
        extends ArrayAdapter<Tweet> {
    private static final String TAG = "ANDROTWEET";
    private List<Tweet> TwitterStatus;
    public boolean[] isSelectedPos;
    private LayoutInflater mInflater;
    private LayoutInflater mInflater2;
    private int selectedCount = 0;

    public TweetAdapter(Context paramContext, int paramInt, List<Tweet> paramList) {
        super(paramContext, paramInt, paramList);
        this.mInflater = LayoutInflater.from(paramContext);
        this.TwitterStatus = paramList;
        this.isSelectedPos = new boolean[500];
    }

    public int getSelectedCount() {
        int count = 0;
        for (int i = 0; i < isSelectedPos.length; i++) {
            if (isSelectedPos[i])
                count += 1;
        }
        selectedCount = count;
        return this.selectedCount;
    }

    @Override
    public View getView(final int position, View convertView, final ViewGroup parent) {
        View row = convertView;

        TweetViewHolder tweetVH;
        if (row == null) {
            row = mInflater.inflate(R.layout.tweets_layout, null);

            tweetVH = new TweetViewHolder();

            tweetVH.txtTweet = (TextView) row.findViewById(R.id.tweet_on_tweets);
            tweetVH.rtTweet = (TextView) row.findViewById(R.id.rtCount_on_tweets);
            tweetVH.favTweet = (TextView) row.findViewById(R.id.favCount_on_tweets);
            tweetVH.timeTweet = (TextView) row.findViewById(R.id.time_on_tweets);
            tweetVH.chkTweet = (CheckBox) row.findViewById(R.id.chkTweet);

            row.setTag(tweetVH);
        } else {
            tweetVH = (TweetViewHolder) row.getTag();
        }
        Tweet tweet = TwitterStatus.get(position);

        tweetVH.chkTweet.setOnClickListener(new CompoundButton.OnClickListener() {

            @Override
            public void onClick(View view) {
                CheckBox c = (CheckBox) view;
                boolean isChecked = (c.isChecked());
                isSelectedPos[position] = isChecked;
            }

        });

        tweetVH.chkTweet.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {

            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                isSelectedPos[position] = isChecked;
                Main.selectedCountChange(getSelectedCount());
            }
        });

//        tweetVH.chkTweet.setText(String.valueOf(tweet.getId()));
        tweetVH.txtTweet.setText(tweet.getTweetText());
        tweetVH.rtTweet.setText(Integer.toString(tweet.getRTcount()));
        tweetVH.favTweet.setText(Integer.toString(tweet.getFAVcount()));
        tweetVH.timeTweet.setText(DateUtils.getRelativeTimeSpanString(tweet.getTime()));
        tweetVH.chkTweet.setChecked(isSelectedPos[position]);

        if (AndroTweetApp.getTweetId()!=null && String.valueOf(tweet.getId()).equals(AndroTweetApp.getTweetId()) && AndroTweetApp.getDaysAgo()<4) {
            tweetVH.chkTweet.setVisibility(View.INVISIBLE);
            isSelectedPos[position]=false;
        }

        return row;
    }

    public class TweetViewHolder {
        CheckBox chkTweet;
        ImageView favImage;
        TextView favTweet;
        ImageView rtImage;
        TextView rtTweet;
        TextView timeTweet;
        TextView txtTweet;

        public TweetViewHolder() {
        }
    }
}

File: app/src/main/java/com/zsk/androtweet/adapters/TweetAdapter.kt
package com.zsk.androtweet.adapters

import android.content.Context
import android.text.format.DateUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.CheckBox
import android.widget.TextView
import com.zsk.androtweet.AndroTweetApp.Companion.daysAgo
import com.zsk.androtweet.AndroTweetApp.Companion.tweetId
import com.zsk.androtweet.Main.Companion.selectedCountChange
import com.zsk.androtweet.Models.Tweet
import com.zsk.androtweet.R

class TweetAdapter(paramContext: Context, paramInt: Int, paramList: List<Tweet>)
    : ArrayAdapter<Tweet?>(paramContext, paramInt, paramList) {
    private val tweetList: List<Tweet> = paramList
    @JvmField
    var isSelectedPos: BooleanArray = BooleanArray(500)
    val selectedCount: Int
        get() {
            var count = 0
            for (i in isSelectedPos.indices) {
                if (isSelectedPos[i]) count += 1
            }
            return count
        }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var row = convertView
        val tweetVH: TweetViewHolder
        if (row == null) {
            val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = layoutInflater.inflate(R.layout.tweets_layout, null)
            tweetVH = TweetViewHolder()
            tweetVH.txtTweet = row.findViewById(R.id.tweet_on_tweets)
            tweetVH.rtTweet = row.findViewById(R.id.rtCount_on_tweets)
            tweetVH.favTweet = row.findViewById(R.id.favCount_on_tweets)
            tweetVH.timeTweet = row.findViewById(R.id.time_on_tweets)
            tweetVH.chkTweet = row.findViewById(R.id.chkTweet)
            row.tag = tweetVH
        } else {
            tweetVH = row.tag as TweetViewHolder
        }
        val tweet = tweetList[position]
        tweetVH.chkTweet!!.setOnClickListener { view ->
            val c = view as CheckBox
            val isChecked = c.isChecked
            isSelectedPos[position] = isChecked
        }
        tweetVH.chkTweet!!.setOnCheckedChangeListener { buttonView, isChecked ->
            isSelectedPos[position] = isChecked
            selectedCountChange(selectedCount)
        }
        //        tweetVH.chkTweet.setText(String.valueOf(tweet.getId()));
        tweetVH.txtTweet!!.text = tweet.tweetText
        tweetVH.rtTweet!!.text = tweet.rTcount.toString()
        tweetVH.favTweet!!.text = tweet.faVcount.toString()
        tweetVH.timeTweet!!.text = DateUtils.getRelativeTimeSpanString(tweet.time)
        tweetVH.chkTweet!!.isChecked = isSelectedPos[position]
        if (tweetId != null && tweet.id.toString() == tweetId && daysAgo < 4) {
            tweetVH.chkTweet!!.visibility = View.INVISIBLE
            isSelectedPos[position] = false
        }
        return row
    }

    inner class TweetViewHolder {
        var chkTweet: CheckBox? = null
        //        var favImage: ImageView? = null
        var favTweet: TextView? = null
        //        var rtImage: ImageView? = null
        var rtTweet: TextView? = null
        var timeTweet: TextView? = null
        var txtTweet: TextView? = null
    }

    companion object {
        private const val TAG = "ANDROTWEET"
    }

}